BurnOutSharp/PackerType/SetupFactory.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Xml;
+
+namespace BurnOutSharp.PackerType
+{
+    public class SetupFactory : IContentCheck, IScannable
+    {
+        /// <inheritdoc/>
+        public bool ShouldScan(byte[] magic) => true;
+
+        /// <inheritdoc/>
+        public string CheckContents(string file, byte[] fileContent, bool includePosition = false)
+        {
+            // "S.e.t.u.p. .F.a.c.t.o.r.y. .i.s. .a. .t.r.a.d.e.m.a.r.k. .o.f. .I.n.d.i.g.o. .R.o.s.e. .C.o.r.p.o.r.a.t.i.o.n"
+            byte[] check = new byte[] { 0x53, 0x00, 0x65, 0x00, 0x74, 0x00, 0x75, 0x00, 0x70, 0x00, 0x20, 0x00, 0x46, 0x00, 0x61, 0x00, 0x63, 0x00, 0x74, 0x00, 0x6F, 0x00, 0x72, 0x00, 0x79, 0x00, 0x20, 0x00, 0x69, 0x00, 0x73, 0x00, 0x20, 0x00, 0x61, 0x00, 0x20, 0x00, 0x74, 0x00, 0x72, 0x00, 0x61, 0x00, 0x64, 0x00, 0x65, 0x00, 0x6D, 0x00, 0x61, 0x00, 0x72, 0x00, 0x6B, 0x00, 0x20, 0x00, 0x6F, 0x00, 0x66, 0x00, 0x20, 0x00, 0x49, 0x00, 0x6E, 0x00, 0x64, 0x00, 0x69, 0x00, 0x67, 0x00, 0x6F, 0x00, 0x20, 0x00, 0x52, 0x00, 0x6F, 0x00, 0x73, 0x00, 0x65, 0x00, 0x20, 0x00, 0x43, 0x00, 0x6F, 0x00, 0x72, 0x00, 0x70, 0x00, 0x6F, 0x00, 0x72, 0x00, 0x61, 0x00, 0x74, 0x00, 0x69, 0x00, 0x6F, 0x00, 0x6E };          
+            if (fileContent.Contains(check, out int position))
+            {
+                string version;
+                // "<?xml"
+                byte[] check2 = new byte[] { 0x3C, 0x3F, 0x78, 0x6D, 0x6C };
+                if (fileContent.Contains(check2, out int position2))
+                {
+                    version = GetVersionXML(fileContent, position2);
+                    if (version == null)
+                        return $"Setup Factory (Unknown Version)" + (includePosition ? $" (Index {position})" : string.Empty);
+                    return $"Setup Factory {version}" + (includePosition ? $" (Index {position}, {position2})" : string.Empty);
+                }
+
+                if (!File.Exists(file))
+                    return null;
+                version = Utilities.GetFileVersion(file);
+                if (version == null)
+                    return $"Setup Factory (Unknown Version)" + (includePosition ? $" (Index {position})" : string.Empty);
+                return $"Setup Factory {version}" + (includePosition ? $" (Index {position})" : string.Empty);
```

Comment:
I see what you were trying to do, but your current code will return no protection at all if the file itself doesn't exist. I don't think that's what you wanted to do, since you already detected the installer at the top level. The below suggested code change still returns the unknown version string if the file doesn't exist to check physically. The utility method already checks for file existence.

```suggestion
                // "<?xml"
                byte[] check2 = new byte[] { 0x3C, 0x3F, 0x78, 0x6D, 0x6C };
                if (fileContent.Contains(check2, out int position2))
                {
                    string version = GetVersionXML(fileContent, position2);
                    if (version == null)
                        return $"Setup Factory (Unknown Version)" + (includePosition ? $" (Index {position})" : string.Empty);
                    
                    return $"Setup Factory {version}" + (includePosition ? $" (Index {position}, {position2})" : string.Empty);
                }
                else
                {
                    string version = Utilities.GetFileVersion(file);
                    if (version == null)
                        return $"Setup Factory (Unknown Version)" + (includePosition ? $" (Index {position})" : string.Empty);
                    
                    return $"Setup Factory {version}" + (includePosition ? $" (Index {position})" : string.Empty);
                }
```